src/groups/mqb/mqbblp/mqbblp_clusterqueuehelper.cpp

```diff
@@ -826,44 +844,44 @@ void ClusterQueueHelper::assignUpstreamSubqueueId(OpenQueueContext* context)
 }
 
 void ClusterQueueHelper::processOpenQueueRequest(
-    const OpenQueueContext& context)
+    const OpenQueueContextSp& context)
 {
     // executed by the cluster *DISPATCHER* thread
 
     // PRECONDITIONS
     BSLS_ASSERT_SAFE(
         d_cluster_p->dispatcher()->inDispatcherThread(d_cluster_p));
-    BSLS_ASSERT_SAFE(isQueueAssigned(*(context.d_queueContext_p)));
+    BSLS_ASSERT_SAFE(isQueueAssigned(*(context->d_queueContext_p)));
```

Comment:
Here, and in other places, we dereference `d_queueContext_p` without checking that it's not null. We rely on the fact that it was correctly set before. We are on the control path, so a few extra checks will not harm the performance. What about adding something like
```suggestion
    BSLS_ASSERT_SAFE(context->d_queueContext_p);
    BSLS_ASSERT_SAFE(isQueueAssigned(*(context->d_queueContext_p)));
```